src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -2755,14 +2760,27 @@ protected void ReceiveFrom_Helper(IPAddress listenOn, IPAddress connectTo)
         {
             using (Socket serverSocket = new Socket(SocketType.Dgram, ProtocolType.Udp))
             {
-                serverSocket.ReceiveTimeout = 500;
+                serverSocket.ReceiveTimeout = 1000;
                 int port = serverSocket.BindToAnonymousPort(listenOn);
 
-                SocketUdpClient client = new SocketUdpClient(_log, serverSocket, connectTo, port);
+                SocketUdpClient client = new SocketUdpClient(_log, serverSocket, connectTo, port, sendNow: false);
+
+                if (NetEventSource.IsEnabled)
+                {
+                    NetEventSource.Info(this, $"SRC:{connectTo} {port} DST:{serverSocket.LocalEndPoint}");
+                }
+
+                // wait for send to finish instead of spawning background task to avoid race condition
```

Comment:
ClientSend is calling SendToAsync and isn't waiting for the send to complete, so technically this comment is only correct if the send completes synchronously.

Reply:
I missed that. for now, the test is passing in my local runs. I can revisit it if this fails again in future.